core/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -256,7 +257,7 @@ public void addIndexStore(String type, Function<IndexSettings, IndexStore> provi
      * @param name Name of the SimilarityProvider
      * @param similarity SimilarityProvider to register
```

Comment:
I think I'd prefer to have an interface for building the similarity. That way you can document what the two `Settings` are.